src/components/pastAnalysis/PastAnalysisTable.tsx

```diff
@@ -172,23 +173,41 @@ const PastAnalysisTable = () => {
   }
 
   const updateRows = (listOfAnalysis: StudyInstanceWithSeries[]) => {
-    const rows: (tableRowsChild | tableRowsParent)[] = []
+    const rows: (tableRowsChild | tableRowsParent)[] = [];
     for (const analysis of listOfAnalysis) {
+      const validAnalyses = analysis.series.filter((series: ISeries) => series.classifications.size > 0);
+      const classifications = validAnalyses[0]?.classifications ? Array.from(validAnalyses[0]?.classifications?.keys()) : [];
```

Comment:
What happens when after filtering out series, `validAnalyses` has 0 items in the array? This could result in `Index out of bound` error.

Please, use optional chaining.
```suggestion
      const classifications = validAnalyses?.[0]?.classifications ? Array.from(validAnalyses?.[0]?.classifications?.keys()) : [];
```